iceoryx_examples/complexdata/README.md

```diff
@@ -0,0 +1,25 @@
+# complexdata
+
+## Introduction
+
+- to implement zero-copy data transfer we use a shared memory approach
+- this requires that every data structure needs to be stored in the shared memory and that
+  they do not use the heap (link to restrictions chapter in overview.md)
+- most of the STL types cannot be used, but we reimplemented some (link to utils readme/cxx)
+- this example shows a) how to send/receive a iox::cxx::vector b) how to send/receive a
+  complex data structure containing some of our STL container surrogates
```

Comment:
The requirement is a bit stronger.
1. Cannot use dynamic memory (I would not write heap, because it is misleading and may have other meanings).
2. Cannot use pointers internally (even if no dynamic memory is used)
3. Cannot reference external data (by index or pointer, since if you do so the external data must be present at the destination as well which adds a lot of complications)